pkg/preemptiontoleration/preemption_toleration.go

```diff
@@ -0,0 +1,422 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package preemptiontoleration
+
+import (
+	"context"
+	"fmt"
+	"sort"
+	"sync"
+	"time"
+
+	v1 "k8s.io/api/core/v1"
+	policy "k8s.io/api/policy/v1beta1"
+	"k8s.io/apimachinery/pkg/runtime"
+	corelisters "k8s.io/client-go/listers/core/v1"
+	policylisters "k8s.io/client-go/listers/policy/v1beta1"
+	schedulinglisters "k8s.io/client-go/listers/scheduling/v1"
+	corev1helpers "k8s.io/component-helpers/scheduling/corev1"
+	"k8s.io/klog/v2"
+	extenderv1 "k8s.io/kube-scheduler/extender/v1"
+	podutil "k8s.io/kubernetes/pkg/api/v1/pod"
+	"k8s.io/kubernetes/pkg/scheduler/apis/config/validation"
+	"k8s.io/kubernetes/pkg/scheduler/framework"
+	dp "k8s.io/kubernetes/pkg/scheduler/framework/plugins/defaultpreemption"
+	"k8s.io/kubernetes/pkg/scheduler/metrics"
+	"k8s.io/kubernetes/pkg/scheduler/util"
+	"sigs.k8s.io/scheduler-plugins/pkg/apis/config"
+
+	schedulerapisconfig "k8s.io/kubernetes/pkg/scheduler/apis/config"
+)
+
+const (
+	// Name of the plugin used in the plugin registry and configurations.
+	Name = "PreemptionToleration"
+)
+
+var (
+	_ framework.PostFilterPlugin = &PreemptionToleration{}
+)
+
+// PreemptionToleration is a PostFilter plugin implements the preemption logic.
+type PreemptionToleration struct {
+	fh                  framework.Handle
+	args                config.PreemptionTolerationArgs
+	podLister           corelisters.PodLister
+	pdbLister           policylisters.PodDisruptionBudgetLister
+	priorityClassLister schedulinglisters.PriorityClassLister
+	clock               util.Clock
+}
+
+var _ framework.PostFilterPlugin = &PreemptionToleration{}
+
+// Name returns name of the plugin. It is used in logs, etc.
+func (pl *PreemptionToleration) Name() string {
+	return Name
+}
+
+// New initializes a new plugin and returns it.
+func New(rawArgs runtime.Object, fh framework.Handle) (framework.Plugin, error) {
+	args, ok := rawArgs.(*config.PreemptionTolerationArgs)
+	if !ok {
+		return nil, fmt.Errorf("got args of type %T, want *PreemptionTolerationArgs", args)
+	}
+
+	if err := validation.ValidateDefaultPreemptionArgs((schedulerapisconfig.DefaultPreemptionArgs)(*args)); err != nil {
```

Comment:
This doesn't fail because it is just declared by NamedType

```go
type PreemptionTolerationArgs schedulerapisconfig.DefaultPreemptionArgs
```

see: https://github.com/everpeace/scheduler-plugins/blob/preemption-toleration-impl/pkg/apis/config/types.go#L152-L155